### PR TITLE
Make phone number optional

### LINKED ISSUE #54: Make phone number optional in the add command.

### PR DESCRIPTION
**Pull request by dasha3412**
Fixes #54

### CODE REVIEW

File: src/test/java/seedu/address/model/person/PhoneTest.java [hunk 1]
```diff
@@ -13,20 +13,16 @@ public void constructor_null_throwsNullPointerException() {
...
-        String invalidPhone = "";
-        assertThrows(IllegalArgumentException.class, () -> new Phone(invalidPhone));
-    }
-
     @Test
     public void isValidPhone() {
         // null phone number
         assertThrows(NullPointerException.class, () -> Phone.isValidPhone(null));
 
+        // phone numbers now option, so can be empty strings
```
Comment (tingxuanp) [inline src/test/java/seedu/address/model/person/PhoneTest.java:21]: Should this be "now optional"?

File: src/main/java/seedu/address/model/person/Phone.java [hunk 2]
```diff
@@ -12,7 +12,9 @@ public class Phone {
 
     public static final String MESSAGE_CONSTRAINTS =
             "Phone numbers should only contain numbers, and it should be at least 3 digits long";
-    public static final String VALIDATION_REGEX = "\\d{3,}";
+
+    // allow either a blank string, whitespaces (will be trimmed) or require all numbers and at least 3 digits long
```
Comment (tingxuanp) [inline src/main/java/seedu/address/model/person/Phone.java:16]: I know the requirement for the phone numbers in AB3 was to be at least 3 digits long but I'm wondering if we should enforce it to be exactly 8 digits  (and have first digit 8/9)? Assuming the wedding planners are based in Singapore... just not sure if others testing the application in the future will pick this up as a bug

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 3]
```diff
@@ -135,6 +144,14 @@ public void parse_optionalFieldsMissing_success() {
         Person expectedPerson = new PersonBuilder(AMY).withTags().build();
         assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + ADDRESS_DESC_AMY,
                 new AddCommand(expectedPerson));
+
+        // blank phone number
+        Person expectedPersonBlankPhone = new PersonBuilder(DOMINIC).withTags().build();
+        assertParseSuccess(parser, NAME_DESC_DOMINIC + PHONE_DESC_DOMINIC + EMAIL_DESC_DOMINIC
+                        + ADDRESS_DESC_DOMINIC, new AddCommand(expectedPersonBlankPhone));
+        // no phone number tag
```
Comment (tingxuanp) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:152]: again minor nit but you could leave a line between 151 and 152 to be consistent with other parts?

**Review by tingxuanp — COMMENTED** [review]
Overall LGTM!

File: src/test/java/seedu/address/model/person/PhoneTest.java (hunk 1 above)
Comment (tingxuanp) [reply src/test/java/seedu/address/model/person/PhoneTest.java:21]: Also the same comment as for the email - would it be better to just put "phone numbers optional" assuming others looking at the code in the future won't know phone numbers were initially compulsory?

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 4]
```diff
@@ -102,10 +107,14 @@ public void parse_repeatedNonTagValue_failure() {
         assertParseFailure(parser, INVALID_EMAIL_DESC + validExpectedPersonString,
                 Messages.getErrorMessageForDuplicatePrefixes(PREFIX_EMAIL));
 
-        // invalid phone
+        // invalid phone - wrong characters
```
Comment (riccoljy) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:110]: Was a bit confused with "wrong characters" but realised this meant "phone number has characters (and not just numbers)". Maybe a slight rephrase would be good?

File: src/test/java/seedu/address/model/person/PhoneTest.java (hunk 1 above)
Comment (riccoljy) [reply src/test/java/seedu/address/model/person/PhoneTest.java:21]: Agree with @tingxuanp about discarding the "now"

**Review by riccoljy — APPROVED** [review]
LGTM -- same nits as @tingxuanp and as previous PR reviews on #74 and #75

**Pull request merged**